src/OrchardCore.Modules/OrchardCore.ContentLocalization/Shortcodes/LocaleShortcodeProvider.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Shortcodes;
+
+namespace OrchardCore.ContentLocalization.Shortcodes
+{
+    public class LocaleShortcodeProvider : IShortcodeProvider
+    {
+        public const string ShortCodeIdentifier = "locale";
+        private static readonly ValueTask<string> Null = new ValueTask<string>((string)null);
+
+        public ValueTask<string> EvaluateAsync(string identifier, Arguments arguments, string content, Context context)
+        {
+            if (identifier != ShortCodeIdentifier)
+            {
+                return Null;
+            }
+
+            var language = arguments.NamedOrDefault("lang")?.ToLower();
+            var argFallback = arguments.NamedOrAt("fallback", 1);
+            // default value of true for the fallback
+            var fallback = argFallback == null ? true : Convert.ToBoolean(argFallback); ;
+            var currentCulture = CultureInfo.CurrentCulture;
```

Comment:
`CurrentCulture` or `CurrentUICulture`. I know we set both but I think for views we need to use the later.

Reply:
Use `CurrentUICulture` as Seb mentioned